src/call/mod.rs

```diff
@@ -170,16 +175,49 @@ impl Display for RpcStatus {
 
 impl RpcStatus {
     /// Create a new [`RpcStatus`].
-    pub fn new<T: Into<RpcStatusCode>>(code: T, details: Option<String>) -> RpcStatus {
+    pub fn new<T: Into<RpcStatusCode>>(code: T) -> RpcStatus {
+        RpcStatus::with_message(code, String::new())
+    }
+
+    /// Create a new [`RpcStatus`] with given message.
+    pub fn with_message<T: Into<RpcStatusCode>>(code: T, message: String) -> RpcStatus {
+        RpcStatus::with_error_details(code, message, vec![])
+    }
+
+    /// Create a new [`RpcStats`] with code, error_message and error_details.
+    pub fn with_error_details<T: Into<RpcStatusCode>>(
+        code: T,
+        message: String,
+        details: Vec<u8>,
+    ) -> RpcStatus {
         RpcStatus {
-            status: code.into(),
+            code: code.into(),
+            message,
             details,
         }
     }
 
     /// Create a new [`RpcStatus`] that status code is Ok.
     pub fn ok() -> RpcStatus {
-        RpcStatus::new(RpcStatusCode::OK, None)
+        RpcStatus::new(RpcStatusCode::OK)
+    }
+
+    /// Return the instance's error code.
+    #[inline]
+    pub fn error_code(&self) -> RpcStatusCode {
```

Comment:
It seems that this is not necessarily an error code in that the code could be OK.

Reply:
Indeed, I just follow the name of the C++'s version.